memory-e2e/order_service.py

```
@@ -0,0 +1,32 @@
+    cursor.execute("SELECT * FROM orders WHERE id = '" + order_id + "'")
+    return cursor.fetchone()
+
+
+def unit_price(total_cents: int, quantity: int) -> float:
+    # Divide by zero when an order line has no quantity.
+    return total_cents / quantity
+
+
+def api_token(user_id: str) -> str:
+    # MD5 is not a secure hash for a credential.
+    return hashlib.md5(user_id.encode()).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)